internal/storageboxsnapshot/resource.go

```diff
@@ -0,0 +1,286 @@
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hetznercloud/hcloud-go/v2/hcloud"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/util"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/util/experimental"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/util/hcloudutil"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/util/resourceutil"
+)
+
+// ResourceType is the type name of the Hetzner Storage Box Snapshot resource.
+const ResourceType = "hcloud_storage_box_snapshot"
+
+var _ resource.Resource = (*Resource)(nil)
+var _ resource.ResourceWithConfigure = (*Resource)(nil)
+var _ resource.ResourceWithImportState = (*Resource)(nil)
+
+type Resource struct {
+	client *hcloud.Client
+}
+
+func NewResource() resource.Resource {
+	return &Resource{}
+}
+
+// Metadata should return the full name of the resource.
+func (r *Resource) Metadata(_ context.Context, _ resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = ResourceType
+}
+
+// Configure enables provider-level data or clients to be set in the
+// provider-defined Resource type. It is separately executed for each
+// ReadResource RPC.
+func (r *Resource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	experimental.StorageBox.AppendDiagnostic(&resp.Diagnostics)
+
+	var newDiags diag.Diagnostics
+
+	r.client, newDiags = hcloudutil.ConfigureClient(req.ProviderData)
+	resp.Diagnostics.Append(newDiags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Schema should return the schema for this resource.
+func (r *Resource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema.MarkdownDescription = `
+Provides a Hetzner Storage Box Snapshot resource.
+
+See the [Storage Box Snapshots API documentation](https://docs.hetzner.cloud/reference/hetzner#storage-box-snapshots) for more details.
+`
+
+	experimental.StorageBox.AppendNotice(&resp.Schema.MarkdownDescription)
+
+	resp.Schema.Attributes = map[string]schema.Attribute{
+		"storage_box": schema.Int64Attribute{
+			MarkdownDescription: "ID of the Storage Box.",
+			Required:            true,
+			PlanModifiers: []planmodifier.Int64{
+				int64planmodifier.UseStateForUnknown(),
+				int64planmodifier.RequiresReplace(),
+			},
+		},
```

Comment:
About a `_id` suffix.

In other resource e.g. in `server_network` we use `server_id` and `network_id`. 

For server, we use `server_type`, because we should accept ID or Name, but it seem we only accept the name.

What is the pattern that we should follow here? I'd rather only accept an ID, so we don't have to fetch the API ourselves for each snapshot resource. But in that case maybe a id suffix could be better?

Reply:
Ack, I think I only looked at `server`, but nearly all other references (to user-resources) use `_id`.